crates/typed-store/src/raw_key_db_map.rs

```diff
@@ -0,0 +1,498 @@
+        Ok(RawKeyDBMap {
+            inner,
+            _phantom: PhantomData,
+        })
+    }
+
+    /// Puts a key-value pair within a transaction.
+    pub fn put_cf_with_txn(
+        &self,
+        txn: &rocksdb::Transaction<'_, OptimisticTransactionDB>,
+        key: &K,
+        value: &V,
+    ) -> Result<(), TypedStoreError> {
+        let key_bytes = key.serialize()?;
+        let value_bytes =
+            bcs::to_bytes(value).map_err(|e| TypedStoreError::SerializationError(e.to_string()))?;
+        let cf = self.inner.cf()?;
+
+        txn.put_cf(&cf, key_bytes, value_bytes)
+            .map_err(|e| TypedStoreError::RocksDBError(e.to_string()))
+    }
+
+    /// Deletes a key within a transaction.
+    pub fn delete_cf_with_txn(
+        &self,
+        txn: &rocksdb::Transaction<'_, OptimisticTransactionDB>,
+        key: &K,
+    ) -> Result<(), TypedStoreError> {
+        let key_bytes = key.serialize()?;
+        let cf = self.inner.cf()?;
+
+        txn.delete_cf(&cf, key_bytes)
+            .map_err(|e| TypedStoreError::RocksDBError(e.to_string()))
+    }
+
+    /// Gets a value for update within a transaction, the commit will fail if the value is updated
+    /// outside the transaction.
+    pub fn get_for_update_cf(
+        &self,
+        txn: &rocksdb::Transaction<'_, OptimisticTransactionDB>,
+        key: &K,
+    ) -> Result<Option<V>, TypedStoreError> {
+        let key_bytes = key.serialize()?;
+        let cf = self.inner.cf()?;
+
+        let value_bytes = txn
+            .get_for_update_cf(&cf, &key_bytes, true)
+            .map_err(|e| TypedStoreError::RocksDBError(e.to_string()))?;
+
+        match value_bytes {
+            Some(bytes) => {
+                let value = bcs::from_bytes(&bytes)
+                    .map_err(|e| TypedStoreError::SerializationError(e.to_string()))?;
+                Ok(Some(value))
+            }
+            None => Ok(None),
+        }
+    }
+
+    /// Gets a value within a transaction.
+    pub fn get_cf_with_txn(
+        &self,
+        txn: &rocksdb::Transaction<'_, OptimisticTransactionDB>,
+        key: &K,
+    ) -> Result<Option<V>, TypedStoreError> {
+        let key_bytes = key.serialize()?;
+        let cf = self.inner.cf()?;
+
+        let value_bytes = txn
+            .get_cf(&cf, &key_bytes)
+            .map_err(|e| TypedStoreError::RocksDBError(e.to_string()))?;
+
+        match value_bytes {
+            Some(bytes) => {
+                let value = bcs::from_bytes(&bytes)
+                    .map_err(|e| TypedStoreError::SerializationError(e.to_string()))?;
+                Ok(Some(value))
+            }
+            None => Ok(None),
+        }
+    }
+
+    /// Reads a range of key-value pairs within a transaction using efficient range bounds.
+    /// Returns key-value pairs in the range [begin, end).
+    ///
+    /// This is more efficient than prefix-based iteration for range queries.
+    pub fn read_range(
```

Comment:
question: DBMap already has iterator support https://github.com/MystenLabs/walrus/blob/c4db4518ac8bd8876446a290d23be1ffe2dfcfb5/crates/typed-store/src/rocks.rs#L2189-L2218, can we use it here?

Reply:
Ah, I missed the transaction part. Maybe mention this in the function name like other functions?